src/containerapp/azext_containerapp/custom.py

```diff
@@ -0,0 +1,954 @@
+                                existing_container['env'] = []
+                            _add_or_update_env_vars(existing_container['env'], new_container['env'])
+                        if 'command' in new_container and new_container['command']:
+                            existing_container['command'] = new_container['command']
+                        if 'args' in new_container and new_container['args']:
+                            existing_container['args'] = new_container['args']
+                        if 'resources' in new_container and new_container['resources']:
+                            if 'cpu' in new_container['resources'] and new_container['resources']['cpu'] is not None:
+                                existing_container['resources']['cpu'] = new_container['resources']['cpu']
+                            if 'memory' in new_container['resources'] and new_container['resources']['memory'] is not None:
+                                existing_container['resources']['memory'] = new_container['resources']['memory']
+
+                # If not updating existing container, add as new container
+                if not updating_existing_container:
+                    current_containerapp_def["properties"]["template"]["containers"].append(new_container)
+
+    # Traffic Weights
+
+    # Secrets
+
+    # Registries
+
+    # Scale rules
+
+    # Source Controls
+
+    '''
+    try:
+        r = ContainerAppClient.create_or_update(
+            cmd=cmd, resource_group_name=resource_group_name, name=name, container_app_envelope=current_containerapp_def, no_wait=no_wait)
+
+        if "properties" in r and "provisioningState" in r["properties"] and r["properties"]["provisioningState"].lower() == "waiting" and not no_wait:
+            logger.warning('Containerapp creation in progress. Please monitor the creation using `az containerapp show -n {} -g {}`'.format(
+                name, resource_group_name
+            ))
+
+        return r
+    except Exception as e:
+        handle_raw_exception(e)
+
+
+def create_containerapp_yaml(cmd, name, resource_group_name, file_name, no_wait=False):
+    yaml_containerapp = process_loaded_yaml(load_yaml_file(file_name))
+    if type(yaml_containerapp) != dict:
+        raise ValidationError('Invalid YAML provided. Please see https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec#examples for a valid containerapps YAML spec.')
+
+    if not yaml_containerapp.get('name'):
+        yaml_containerapp['name'] = name
+    elif yaml_containerapp.get('name').lower() != name.lower():
+        logger.warning('The app name provided in the --yaml file "{}" does not match the one provided in the --name flag "{}". The one provided in the --yaml file will be used.'.format(
+            yaml_containerapp.get('name'), name))
+    name = yaml_containerapp.get('name')
+
+    if not yaml_containerapp.get('type'):
+        yaml_containerapp['type'] = 'Microsoft.App/containerApps'
```

Comment:
generic question wherever I see path should this OS agnostic?

Reply:
From my understanding resource type should always be in the format "Microsoft.RP/resourceType"